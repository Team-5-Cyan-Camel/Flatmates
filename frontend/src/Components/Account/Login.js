import React from "react";
import { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import axios from "axios";

import "bootstrap/dist/css/bootstrap.min.css";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";

const Login = () => {
  let [user, setUser] = useState("");
  let [password, setPassword] = useState("");
  const history = useHistory();

  useEffect(() => {
    axios
      .get("/user")
      .then((res) => {
        console.log(res.data);
        if (res.data.roomCode !== null) {
          history.push("/room/" + res.data.roomCode);
        } else {
          history.push("/code");
        }
      })
      .catch(function (error) {
        console.log(error);
      });
  }, []);

  const login = (e) => {
    e.preventDefault();
    //  check if user account exist (axios)

    const userIn = {
      username: user,
      password: password,
    };

    axios
      .post("user/login", userIn)
      .then((res) => {
        let roomcode = res.data.roomCode;
        if (roomcode !== null) {
          history.push("/room/" + roomcode);
        } else {
          history.push("/code");
        }
      })
      .catch(function (error) {
        console.log(error);
      });
  };

  return (
    <>
      <Form style={{ marginLeft: "1em", marginRight: "1em" }}>
        <input
          class="AccountInputField"
          type="text"
          name="username"
          placeholder="Username"
          value={user}
          onChange={(e) => setUser(e.target.value)}
        />
        <br></br>
        <input
          class="AccountInputField"
          type="password"
          id="password"
          name="password"
          placeholder="Password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <br></br>
      </Form>
      <Button className="GoButton" onClick={login}>
        Login
      </Button>
    </>
  );
};

export default Login;
