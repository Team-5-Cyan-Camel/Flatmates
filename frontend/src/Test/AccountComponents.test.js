import React from "react";
import renderer from "react-test-renderer";
import Login from "./../Components/Account/Login";

test("Login box renders correctly", () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
