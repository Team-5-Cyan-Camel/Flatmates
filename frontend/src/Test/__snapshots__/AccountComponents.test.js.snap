// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login box renders correctly 1`] = `
Array [
  <form
    className=""
    style={
      Object {
        "marginLeft": "1em",
        "marginRight": "1em",
      }
    }
  >
    <input
      className="AccountInputField"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <br />
    <input
      className="AccountInputField"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <br />
  </form>,
  <button
    className="GoButton btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Login
  </button>,
]
`;
